src/Repository.cc

```diff
@@ -594,6 +594,27 @@ bool Repository::branch_up_to_date(const std::string& branch_name)
 }
 #endif
 
+void Repository::checkout(const std::string& branch_name, const std::vector<std::string>& paths)
```

Comment:
Signature different from header file where it is 
```c++
    /**
     * Checkout a branch with options
    */
    void checkout(const std::string& branch_name, const std::string& paths = "*");
```
(vector versus signle string)

Reply:
I changed this for my tests to

```diff
diff --git a/include/libgit4cpp/Repository.h b/include/libgit4cpp/Repository.h
index c0f7faf..5b36f0f 100644
--- a/include/libgit4cpp/Repository.h
+++ b/include/libgit4cpp/Repository.h
@@ -239,7 +239,7 @@ public:
     /**
      * Checkout a branch with options
     */
-    void checkout(const std::string& branch_name, const std::string& paths = "*");
+    void checkout(const std::string& branch_name, const std::vector<std::string>& paths = {"*"});
 
     /**
      * Remove all entries from the index under a given directory.
```